rest/route/github_comments.go

```diff
@@ -0,0 +1,56 @@
+package route
+
+import (
+	"fmt"
+
+	"github.com/evergreen-ci/evergreen/model/patch"
+	"github.com/google/go-github/v52/github"
+)
+
+type githubComments struct {
+	baseUrl string
+}
+
+func newGithubComments(baseUrl string) *githubComments {
```

Comment:
Small nugget of wisdom - if you don't have a specific reason to use a pointer, generally try to avoid it. Pointers introduce more surface area for nils and panics, and historically, this fueled a paranoid mentality of "but what if it's a nil pointer??" and resulted in lots of extra panic safety checks that might not actually be needed. You can see some examples of this in the `pr` object, where we use `Get<Field>` everywhere because it's full of pointers and we're not sure what fields GitHub is guaranteed to send us/which ones have already been nil checked.

On a further note about pointers and methods - even if some of its method definitions use a pointer receiver (like `(g * githubComments) existingPatches`), it's still okay to return the struct from constructors - Go is smart enough to figure out the magic of working on the pointer to the struct rather than copying the struct with each method. It seems like none of this needs to use pointers, so just returning the struct as-is from `newGithubComments` should be good enough.

Reply:
Yeah... I even stored it as a struct instead of a pointer in the gh (githubHookApi) struct. Moved to not use a pointer